WordPress/src/future/res/PluginsCheckerWPOrg.java

```diff
@@ -0,0 +1,190 @@
+package org.wordpress.android.ui.accounts.helpers;
+
+import android.text.TextUtils;
+import android.webkit.URLUtil;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wordpress.android.util.AppLog;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PluginsCheckerWPOrg {
+    private final static String BB_PLUGINS_LIST_URL = "https://raw.githubusercontent.com/wordpress-mobile/app-blocking-plugins/master/xmlrpc-plugins.json";
+
+    // Do not use the WP-APP user agent. Requests could be blocked if made from our app UA.
+    private final static String USER_AGENT = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.116 Safari/537.36";
+
+    /** Socket timeout in milliseconds for the requests */
+    private static final int REQUEST_TIMEOUT_MS = 30000;
+
+    String mOriginalURL;
+
+    public PluginsCheckerWPOrg(String url) {
+        mOriginalURL = url;
+    }
+
+    public List<Plugin> checkForPlugins() {
+        String responseHTML = downloadPluginsList();
+        if (TextUtils.isEmpty(responseHTML)) {
+            AppLog.w(AppLog.T.NUX, "Without the list we can't check if the host has some BB plugins installed on it.");
+            return null;
+        }
+
+        JSONArray listOfPlugins;
+        try {
+            listOfPlugins = new JSONArray(responseHTML);
+        } catch (JSONException e) {
+            AppLog.e(AppLog.T.NUX, "Error while parsing the list of plugins returned from the server.", e);
+            return null;
+        }
+
+        // we have the list. Start the process of checking for plugins
+        String baseURL = getBaseURL(mOriginalURL);
+
+        if (!baseURL.contains("/plugins")) {
+            baseURL = baseURL + "/wp-content/plugins/";
```

Comment:
This might be a stupid question as I don't know how this class is going to be used, but is there any way the url contains `/wp-content` but not `/plugins`?